cloudslang-content-verifier/src/main/java/io/cloudslang/lang/tools/build/SlangBuilder.java

```diff
@@ -124,30 +140,56 @@ IRunTestResults runTests(Map<String, Executable> contentSlangModels,
         log.info("Found " + testCases.size() + " tests");
         IRunTestResults runTestsResults;
 
-        runTestsResults = processRunTests(projectPath, testSuites, bulkRunMode, compiledFlows, testCases);
+        BuildModeConfig buildModeConfig = createBuildModeConfig(buildMode, changedFiles, allTestedFlowModels);
+
+        runTestsResults = processRunTests(projectPath, testSuites, bulkRunMode, compiledFlows, testCases, buildModeConfig);
 
         runTestsResults.addExceptions(compilationResult.getExceptions());
         addCoverageDataToRunTestsResults(contentSlangModels, testFlowModels, testCases, runTestsResults);
         return runTestsResults;
     }
 
-    IRunTestResults processRunTests(String projectPath, List<String> testSuites, BulkRunMode bulkRunMode, Map<String, CompilationArtifact> compiledFlows, Map<String, SlangTestCase> testCases) {
+    private BuildModeConfig createBuildModeConfig(SlangBuildMain.BuildMode buildMode, Set<String> changedFiles, Map<String, Executable> allTestedFlowModels) {
+        BuildModeConfig buildModeConfig;
+        switch (buildMode) {
```

Comment:
if else if is a bit shorter in lines of code then switch, no need to change this.